x-pack/plugins/transform/public/app/common/validators.ts

```diff
@@ -5,10 +5,47 @@
  */
 
 /**
- * Validates time delay input.
+ * Validates continuous mode time delay input.
  * Doesn't allow floating intervals.
  * @param value User input value.
  */
-export function delayValidator(value: string): boolean {
+export function continuousModeDelayValidator(value: string): boolean {
   return value.match(/^(0|\d*(nanos|micros|ms|s|m|h|d))$/) !== null;
 }
+
+/**
+ * Validates transform frequency input.
+ * Allows time units of s/m/h only.
+ * Must be above 0 and only up to 1h.
+ * @param value User input value.
+ */
+export const transformFrequencyValidator = (value: string): boolean => {
+  if (typeof value !== 'string' || value === null) {
+    return false;
+  }
+
+  // split string by groups of numbers and letters
+  const regexStr = value.match(/[a-z]+|[^a-z]+/gi);
+
+  return (
+    // only valid if one group of numbers and one group of letters
+    regexStr !== null &&
+    regexStr.length === 2 &&
+    // only valid if time unit is one of s/m/h
+    ['s', 'm', 'h'].includes(regexStr[1]) &&
+    // only valid if number is between 1 and 59
+    parseInt(regexStr[0], 10) > 0 &&
```

Comment:
This assumes a user will always enter '1 hour' as `1h` and not say `3600s` or `60m`. Can the duration of the interval be validated, rather than the digits?

![image](https://user-images.githubusercontent.com/7405507/93209178-a32ae680-f755-11ea-97a4-893747b582d1.png)

Reply:
Improved validation and updated unit tests in 7edc1bdd17b.